hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -678,23 +679,50 @@ public boolean isDereferencedBySubclassProperty() {
 
 	// ParameterContainer impl ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 	private List<ParameterSpecification> embeddedParameters;
+	private List<ParameterSpecification> embeddedFilterParameters;
 
 	@Override
 	public void addEmbeddedParameter(ParameterSpecification specification) {
-		if ( embeddedParameters == null ) {
-			embeddedParameters = new ArrayList<ParameterSpecification>();
+		if ( specification instanceof DynamicFilterParameterSpecification ) {
+			if ( embeddedFilterParameters == null ) {
+				embeddedFilterParameters = new ArrayList<ParameterSpecification>();
+			}
+			embeddedFilterParameters.add( specification );
+		}
+		else {
+			if ( embeddedParameters == null ) {
+				embeddedParameters = new ArrayList<ParameterSpecification>();
+			}
+			embeddedParameters.add( specification );
+
 		}
-		embeddedParameters.add( specification );
 	}
 
 	@Override
 	public boolean hasEmbeddedParameters() {
-		return embeddedParameters != null && ! embeddedParameters.isEmpty();
+		final boolean b = (embeddedParameters != null && !embeddedParameters.isEmpty())
```

Comment:
The temporary variable can be omitted and we can return the boolean expression directly.